app/models/invoice.rb

```diff
@@ -544,8 +542,13 @@ def id_sufix
     friendly_id.sub(/(.+)-/, '')
   end
 
+  after_commit :set_friendly_id, on: :create
   after_commit :update_counter, on: :update
 
+  def set_friendly_id
```

Comment:
Couldn't be a private method?

Reply:
Not necessarily. It is idempotent.

Reply:
I'll make it private then. Actually, that may be good since there's no point in using the method if the record is not persisted.